File: KevinRene/src/cons/ql/ast/expression/literal/Literal.java
package cons.ql.ast.expression.literal;

import cons.ql.ast.Expression;

public abstract class Literal<T> extends Expression {
	
	private final T value;
	
	public Literal(T value) {
		this.value = value;
	}
	
	public T getValue() {
		return this.value;
	}
	
	public String toString() {
		return value.toString();
	}
}

File: CarlaErik/UvA.SoftCon.Questionnaire/QL/AST/Model/Expressions/Literals/Literal.cs
using UvA.SoftCon.Questionnaire.Common.AST.Model;

namespace UvA.SoftCon.Questionnaire.QL.AST.Model.Expressions.Literals
{
    public abstract class Literal : Expression
    {
        public string Value
        {
            get;
            private set;
        }

        public abstract bool IsValid { get; }

        protected Literal(string value, TextPosition position)
            : base(position)
        {
            Value = value;
        }

        public override string ToString()
        {
            return Value.ToString();
        }
    }

    public abstract class Literal<T> : Literal
    {
        protected Literal(string value, TextPosition position)
            : base(value, position) { }

        public abstract T GetValue();
    }
}
